### PR TITLE
Add result tab flag to CodePen embeds

### PR DESCRIPTION
**Pull request by MattIPv4**
## Type of Change

- **Markdown-It Plugins:** CodePen embeds

## What issue does this relate to?

N/A

### What should this PR do?

Adds support for a new `result` tab flag (the default), which can optionally be combined with any of the existing tab flags (`html`, `css`, `js`) to enable a split view in the embed.

This also adds support for a `light` flag (that does nothing, as `light` was already the default theme).

### What are the acceptance criteria?

New flag added and works as expected, can be passed on its own or combined with an existing tab flag. Resulting embed is as expected, either defaulting to just the result view or split if combined with another tab.

Tests are added to cover all potential uses of the new tab flag.

### CODE REVIEW

File: rules/embeds/codepen.js [hunk 1]
```diff
@@ -103,8 +107,11 @@ module.exports = md => {
         // Defines if the embed should be editable
         const editable = flags.includes('editable');
 
-        // Defines the default tab
-        const tab = [ 'html', 'css', 'js' ].find(t => flags.includes(t)) || 'result';
+        // Defines the default tab (preferring based on the order in the tabs array)
+        let tab = tabs.find(t => flags.includes(t)) || 'result';
+
+        // If the result tab was request, it can be combined with any other tab
```
Comment (MattIPv4) [inline rules/embeds/codepen.js:113]:
```suggestion
        // If the result tab was requested, it can be combined with any other tab
```

**Review by xamthor — DISMISSED** [review]
Pulled Locally and Ran Tests

LGTM

File: rules/embeds/codepen.test.js [hunk 2]
```diff
@@ -84,22 +100,70 @@ it('handles codepen embeds with js tab flag', () => {
...
+it('handles codepen embeds with html and css tab flags (preferring html)', () => {
     expect(md.render('[codepen AlbertFeynman gjpgjN css html]')).toBe(`<p class="codepen" data-height="256" data-theme-id="light" data-default-tab="html" data-user="AlbertFeynman" data-slug-hash="gjpgjN" style="height: 256px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;">
     <span>See the Pen <a href="https://codepen.io/AlbertFeynman/pen/gjpgjN">gjpgjN by AlbertFeynman</a> (<a href="https://codepen.io/AlbertFeynman">@AlbertFeynman</a>) on <a href='https://codepen.io'>CodePen</a>.</span>
 </p>
 <script async defer src="https://static.codepen.io/assets/embed/ei.js" type="text/javascript"></script>
 `);
 });
 
-it('handles codepen embeds with three tab flags (preferring html, then css, then js)', () => {
+it('handles codepen embeds with html, css and js tab flags (preferring html)', () => {
```
Comment (davidcarlsonberg) [inline rules/embeds/codepen.test.js:111]:
```suggestion
it('handles codepen embeds with html, css, and js tab flags (preferring html)', () => {
```

**Review by davidcarlsonberg — DISMISSED** [review]
Lgtm, one tiny Oxford comma nit.

**Review by davidcarlsonberg — APPROVED** [review]

**Pull request merged**